benchmark/main.cpp

```diff
@@ -140,21 +137,43 @@ void benchmark(size_t colSize, int colCount, int threadCount, int iterations, bo
             (*thread).join();
         }
 
-        if (!cache || i > 0) {
-            long long int time = accumulate(threadTimes.begin(), threadTimes.end(), (long long int) 0) / threadCount;
-            times.push_back(time);
-        }
-        threadTimes.clear();
-        threadTimes.resize(threadCount);
-
         while (!threads.empty()) {
             delete threads.back();
             threads.pop_back();
         }
         threadFlag = false;
     }
 
-    printResults<T>(times, colSize, threadCount, cache);
+    size_t startIndex = 0;
```

Comment:
I simply moved it from the cache iteration the way it was before. As far as I could tell, the cache was being cleared, followed by a dry-run of reading without using the time measured. The reason I'm not using a for loop with 2 iterations was that I thought it would make the code more obtuse than saving 10 lines was worth.